### PR TITLE
chore(formatter): add formatter to build pipeline

### PR DESCRIPTION
**Pull request by mpiercy827**
### Changes

- [x] Adds Elixir formatter to the `.travis.yml` build pipeline.
- [x] Formats all existing files.
- [x] Updates Elixir version to 1.8 (since the formatter requires >= 1.6, and having updated versions is nice).
- [x] Yeet.

### CODE REVIEW

**Comment by itsachen** [comment]
Oh hi @mpiercy827! Thanks for the submission!

File: .travis.yml [hunk 1]
```diff
@@ -18,6 +18,7 @@ script:
   - mix coveralls.travis --max-cases 1
   - mix credo
   - mix dialyzer --halt-exit-status
+  - mix format --check-formatted
```
Comment (itsachen) [inline .travis.yml]: Could we change https://github.com/lob/lob-elixir/blob/master/CONTRIBUTING.md to mention formatting as well?

**Comment by mpiercy827** [comment]
Oh hai @itsachen! It looks like Travis is erroring due to a missing API Key, is there a test API key that we're using somewhere?

**Comment by itsachen** [comment]
@mpiercy827 I'll check on that

**Comment by itsachen** [comment]
An update on the build - we omit the API key for PRs coming from a forked repo for security purposes. I've set up a branch on this repo with these changes and kicked off a build: https://travis-ci.org/lob/lob-elixir/builds/557940000

**Comment by itsachen** [comment]
@mpiercy827 Looks like your build is 🍏

File: .travis.yml [hunk 2]
```diff
@@ -3,12 +3,11 @@ language: elixir
 sudo: false
 
 elixir:
```
Comment (itsachen) [inline .travis.yml]: Could we add 1.6.x, 1.7.x to this as well?

**Comment by itsachen** [comment]
To recap conversations had offline:
- Lets not bump the min version of Elixir in this PR. We should do it soon but at a separate schedule.
- We'll need to update the TravisCI config to skip running format for incompatible versions (see https://github.com/Nebo15/confex/blob/master/.travis.yml for an example)

With that we should be good to go/merge!

**Comment by mpiercy827** [comment]
Going to close this out and open up a new PR

**Pull request closed without merging**